src/yupSchema.js

```diff
@@ -181,9 +188,14 @@ const getOptions = (field) => {
 const getYupSchema = ({ inputType, ...field }) => {
   const jsonType = getJsonTypeInArray(field.jsonType);
 
-  if (field.options?.length > 0) {
+  if (field.options?.length > 0 && field.jsonType !== 'number') {
+    const optionValues = getOptions(field);
+    return yupSchemas.radioOrSelectString(optionValues);
+  }
+
+  if (field.options?.length > 0 && field.jsonType === 'number') {
```

Comment:
Hi @brennj and @sandrina-p, we just test this against our form. It does not seem to handle ["number", null].